js/modules/popupRegistration.js

```diff
@@ -0,0 +1,93 @@
+const popupLinks = document.querySelectorAll(".popup-link");
+const registerModalOverlay = document.querySelector(".register__modal_overlay");
+const registerModalContent = registerModalOverlay.querySelector(".register__modal_content");
+const registerModalGratitude = registerModalOverlay.querySelector(".register__modal_gratitude");
+const closeBtn = registerModalOverlay.querySelector(".register__close_button");
+const form = registerModalOverlay.querySelector(".register__form");
+const modalSubmitBtn = registerModalOverlay.querySelector(".modal_form_button");
+const REOPEN_DELAY = 800;
+const SHOW_TIME = 3000;
+const KEYCODE = {
+	ESC: "Escape"
+};
+let isOpen = true;
+form.addEventListener("submit", submitHandler);
+registerModalOverlay.addEventListener("click", emptyPlaceCloseHandler);
+closeBtn.addEventListener("click", closeBtnHandler);
+
+popupLinks.forEach( popupLink => {
+	popupLink.addEventListener("click", function(){
+		popupOpen();
+		addButtonContent(popupLink);
+	})
+});
+
+function popupOpen(){
+	if(!isOpen) return;
+
+	registerModalOverlay.classList.add("open");
+	registerModalContent.classList.add("open");
+	disablePageScroll();
+	isOpen = false;
+
+	document.addEventListener("keydown", keyDownHandler);
+
+	setTimeout(() => isOpen = true, REOPEN_DELAY);
+}
+
+function emptyPlaceCloseHandler (event){
+	if(event.target === registerModalOverlay){
+		popupClose(registerModalContent);
+	}
+}
+
+function closeBtnHandler(){
+	popupClose(registerModalContent);
+}
+
+function keyDownHandler(event){
+	if(event.code === KEYCODE.ESC){
+		popupClose(registerModalContent);
+	}
+}
+
+function submitHandler(event){
+	event.preventDefault();
+	registerModalContent.classList.remove("open");
+	registerModalGratitude.classList.add("open");
+	isOpen = false;
+	setTimeout(()=>{
```

Comment:
remove it and give a user an opportunity to close the thanks modal window by clicking a close button when he wants it

Reply:
Done.